deploy/libproject/jirarelease.rb

```diff
@@ -0,0 +1,138 @@
+# Wraps the jira-ruby library to support creating and appending to All of Us
+# release tickets. Requires basic auth credentials. The initial ticket is
+# populated with a formatted git commit log since the last version.
+#
+# Note, this script is a Ruby analog to the RDR python release scripts:
+# - https://github.com/all-of-us/raw-data-repository/blob/master/ci/release_notes.py
+# - https://github.com/all-of-us/raw-data-repository/blob/master/rest-api/tools/update_release_tracker.py
+
+require "open3"
+require "jira-ruby"
+
+require_relative "../../aou-utils/utils/common"
+
+
+REPO_BASE_URL = "https://github.com/all-of-us/workbench"
+
+# Formatting for release notes in JIRA comments.
+# Note that JIRA auto-linkifies JIRA IDs, so avoid using commit message text in a link.
+LOG_LINE_FORMAT = "--format=*   [%aN %ad|" + REPO_BASE_URL + "/commit/%h] %s"
+# Overall release notes template.
+RELEASE_NOTES_T = """h1. Release Notes for %{current}
+h2. deployed to %{project}, listing changes since %{prev}
+%{history}
+"""
+
+JIRA_INSTANCE_URL = "https://precisionmedicineinitiative.atlassian.net/"
+JIRA_PROJECT_NAME = "PD"
+
+def linkify_pull_request_ids(text)
+  # Converts all substrings like "(#123)" to links to pull requests.
+  return text.gsub(
+      /\(#([0-9]+)\)/,
+      '([#\1|' + REPO_BASE_URL + '/pull/\1])')
+end
+
+def get_release_notes_between_tags(project, from_tag, to_tag)
+  """Formats release notes for JIRA from commit messages, between the two tags."""
+  commit_messages = Common.new.capture_stdout(
+    ['git', 'log', "#{from_tag}..#{to_tag}", LOG_LINE_FORMAT])
+  if not commit_messages
+    raise RuntimeError.new "failed to retrieve commits"
+  end
+
+  return RELEASE_NOTES_T % {
+    :current => to_tag,
+    :project => project,
+    :prev => from_tag,
+    :history => linkify_pull_request_ids(commit_messages),
+  }
+end
+
+def format_jira_error(e)
+  return "JIRA request failed with #{e.response.code} #{e.response.message}: " +
+         e.response.body
+end
+
+class JiraReleaseClient
+  def initialize(username, password)
+    # Set :http_debug => true to see outgoing JIRA requests
+    @client = JIRA::Client.new({
+      :site => JIRA_INSTANCE_URL,
+      :context_path => "",
+      :username => username,
+      :password => password,
+      :auth_type => :basic,
+    })
+  end
+
+  def self.from_gcs_creds(project)
+    gcs_uri = "gs://#{project}-credentials/jira-login.json"
```

Comment:
Make sure to include adding this file to the credentials bucket as a part of the new environment setup steps. (Or automate that.)

Reply:
Done